e2e.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Exit immediately if a command exits with a non-zero status
+set -e
+
+# Function to perform cleanup
+cleanup() {
+  echo "Cleaning up..."
+  # Remove generated folders and files
+  rm -rf e2e-tests e2e-tests.zip site_import.zip site_import "$CODE_VERSION" "$CODE_VERSION.zip"
+}
+
+# Trap EXIT signal to run cleanup when the script exits, regardless of success or failure
+trap cleanup EXIT
+
+# Load environment variables from .env file
+if [ -f .env ]; then
+  echo "Loading environment variables from .env file"
+  export $(grep -v '^#' .env | xargs)
+fi
+
+# Ensure CODE_VERSION is set
+if [ -z "$CODE_VERSION" ]; then
+  echo "ERROR: CODE_VERSION is not set. Please define it in your .env file or export it in your shell."
+  exit 1
+fi
+
+echo "Using CODE_VERSION: $CODE_VERSION"
+
+echo "Linting JavaScript..."
+npm run lint:js
+
+echo "Running unit tests..."
+npm run test
+
+echo "Compiling JavaScript..."
+npm run compile:js
+
+echo "Compiling SCSS..."
+npm run compile:scss
+
+echo "Preparing code for deployment..."
+mkdir -p "$CODE_VERSION"
+cp -R cartridges/* "$CODE_VERSION"/
+zip -r "$CODE_VERSION.zip" "$CODE_VERSION"
```

Comment:
You can polish things like that: instead of displaying the verbose zip logs which create a lot of noise around other logs, you can just log that you are zipping the cartridges/ directory.